jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1364,6 +1372,9 @@ public void failed(Throwable failure)
             boolean completeStream;
             try (AutoLock ignored = _request._lock.lock())
             {
+                request = _request;
+                if (request._httpChannel != null && request._httpChannel._error != null && TypeUtil.isAssociated(request._httpChannel._error.getCause(), failure))
+                    return;
```

Comment:
Need a comment here about the fact we are being forgiving of lazy applications that doubly report a failure.
```suggestion
                // Be forgiving of applications that report a failure that is already known by treating as a noop.
                if (request._httpChannel != null && request._httpChannel._error != null && TypeUtil.isAssociated(request._httpChannel._error.getCause(), failure))
                    return;
```